### ISSUE TITLE
Coding Style - Add indentation within a namespace 

### ISSUE THREAD

**Issue by hodoulp** [AcademySoftwareFoundation/OpenColorIO#612]
After several pull requests, we would like to raise a coding style concern around the indentation within a namespace. The [OCIO Coding Style](http://opencolorio.org/developers/coding_guidelines.html) mentions that there is no indentation in a namespace. While coding new features we discovered that the rule introduces some readability complexity because it flattens the code.

Below is an example using a code snippet from the coding style:
```
OCIO_NAMESPACE_ENTER
{

namespace
{

int MungeMyNumbers(int a, int b)
{
    int x = a + b;
    
    if (a == 0 || b==0)
    {
        x += 1;
        x += 2;
    }
    else
    {
        for (int i=0; i<16; ++i)
        {
            x += a * i;
        }
    }
    return x;
}

} // anonymous

void DoSomething()
{
    const int res = MungeMyNumbers(1, 4);
    …
}

}
```
Adding indentations ease the reading of the source code:
```
OCIO_NAMESPACE_ENTER
{
    namespace
    {
        int MungeMyNumbers(int a, int b)
        {
            int x = a + b;
    
            if (a == 0 || b==0)
            {
                x += 1;
                x += 2;
            }
            else
            {
                for (int i=0; i<16; ++i)
                {
                    x += a * i;
                }
            }
            return x;
        }
    } // anonymous

    void DoSomething()
    {
        const int res = MungeMyNumbers(1, 4);
        …
    }
}
```

Most of the existing code is currently adding indentations. So, we propose to change the coding style to have indentation within a namespace.

Please let us know if you agree to change that rule in the OCIO Coding Style.

**Comment by michdolan** [comment]
I also prefer NOT indenting namespaces. Indenting does ease readability when scanning through code, but  it also shortens available characters if you're following a stricter line-length in the agreed upon code style (OCIO source *generally* limits to 80 chars). 

BUT, as you said @hodoulp , much of the existing code does indent namespaces, and I would support sticking with whatever the majority of the code is already doing, since the biggest style rule tends to be following the existing style.

+1 for using clang-format

**Comment by hodoulp** [comment]
Agree that most of the coding style rules are based on personal preferences from the original devs. However, I tried to present a concrete example of unnecessary complexity.

The fact that most of the existing code is indented is a strong argument to update the coding style. The change could at most add 1-2 indentations (i.e. max of 8 characters), it should not drastically change a line.

Note: Most of the time, the 80-character limit is coming from original printer limitations. With modern IDE's and monitors (and no more need for printings) increase the limit is not an issue. However, that's not the topic of the discussion.

Note: Here, at Autodesk, the synColor coding style is to add indentations (i.e. 2 characters), and few years ago, we moved to a 100-character limit.
